evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -135,9 +146,15 @@ def _add_training_data_to_X_Y(self, X, y, X_train, y_train):
         )
         return padded_features, padded_target
 
-    def _drop_time_index(self, X):
+    def _drop_time_index(self, X, y):
         """Helper method to drop the time index column from the data if DateTime Featurizer is not present."""
         if self.should_drop_time_index and self.time_index in X.columns:
+            X_schema = X.ww.schema
+            y_schema = y.ww.schema
+            X = X.set_index(X[self.time_index])
+            y = y.set_axis(X[self.time_index])
+            X.ww.init(schema=X_schema)
+            y.ww.init(schema=y_schema)
```

Comment:
Feels weird to be saving the time index in the data's indices when we're trying to drop it, but the decomposer needs the time index data to decompose successfully. If anyone has suggestions on a better or cleaner way to get around this I am very open to it.

Reply:
Can we drop after fit or transform in the decomposer then? The problem is non-time-series native estimators will fail with a datetime column. Maybe we can amend the logic here to save the time index only when having the decomposer in the pipeline and then dropping the time index later?